pallets/time-release/src/lib.rs

```diff
@@ -450,6 +587,122 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
+
+	fn hold_balance_for(who: &T::AccountId) -> BalanceOf<T> {
+		T::Currency::balance_on_hold(&HoldReason::TimeReleaseScheduledVesting.into(), &who)
+	}
+
+	fn ensure_sufficient_hold_balance(from: &T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+		ensure!(Self::hold_balance_for(&from) >= amount, Error::<T>::InsufficientBalanceToFreeze);
+
+		Ok(())
+	}
+
+	fn validate_and_get_schedule_amount(
+		schedule: &ReleaseScheduleOf<T>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		ensure_valid_release_schedule::<T>(schedule)
+	}
+
+	fn append_release_schedule(
+		recipient: &T::AccountId,
+		schedule: ReleaseScheduleOf<T>,
+	) -> DispatchResult {
+		<ReleaseSchedules<T>>::try_append(recipient, schedule)
+			.map_err(|_| Error::<T>::MaxReleaseSchedulesExceeded)?;
+		Ok(())
+	}
+
+	fn transfer_from_free_balance(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		schedule_amount: BalanceOf<T>,
+	) -> DispatchResult {
+		T::Currency::transfer(from, to, schedule_amount, Preservation::Expendable)?;
+
+		Ok(())
+	}
+
+	/// Transfers the specified amount from the hold balance of the `from` account to the `to` account.
+	///
+	/// # Arguments
+	///
+	/// * `from` - The account from which the funds will be transferred.
+	/// * `to` - The account to which the funds will be transferred.
+	/// * `schedule_amount` - The amount to be transferred.
+	///
+	/// # Errors
+	///
+	/// * `ArithmeticError::Overflow` - If the transfer amount overflows.
+	/// * `Error::InsufficientBalanceToFreeze` - If the hold balance is insufficient.
+	fn transfer_from_hold_balance(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		schedule_amount: BalanceOf<T>,
+	) -> DispatchResult {
+		// Exact - Will fail if the exact amount cannot be transfered
+		// Free - deposit into the free balance but will locked following
+		// Polite - Execute with regular previlages it consideres existential
+		T::Currency::transfer_on_hold(
+			&HoldReason::TimeReleaseScheduledVesting.into(),
+			from,
+			to,
+			schedule_amount,
+			Exact,
+			Free,
+			Polite,
+		)?;
+
+		Ok(())
+	}
+
+	fn ensure_suffient_free_balance(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		amount: BalanceOf<T>,
+	) -> DispatchResult {
+		if to == from {
+			ensure!(T::Currency::balance(&from) >= amount, Error::<T>::InsufficientBalanceToFreeze,);
+		}
+
+		Ok(())
+	}
+
+	fn finalize_vesting_transfer(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		schedule: ReleaseScheduleOf<T>,
+		transfer_fn: fn(&T::AccountId, &T::AccountId, BalanceOf<T>) -> DispatchResult,
+	) -> DispatchResult {
+		let schedule_amount = Self::validate_and_get_schedule_amount(&schedule)?;
+
+		let total_amount = Self::prune_and_get_frozen_balance(to)
+			.checked_add(&schedule_amount)
+			.ok_or(ArithmeticError::Overflow)?;
+
+		transfer_fn(from, to, schedule_amount)?;
+		Self::update_freeze(to, total_amount)?;
+
+		Self::append_release_schedule(to, schedule)?;
+
+		Ok(())
+	}
+
+	fn finalize_vesting_transfer_from_free_balance(
```

Comment:
nit: These last two abstractions seem unnecessary since they are each used in only one place and the single line in each are clear what they're doing.

Reply:
I thought this made the code a little easier to read. Let me know if you want me to change it.